flink-scala/src/main/scala/org/apache/flink/api/scala/GroupedDataSet.scala

```diff
@@ -356,6 +358,44 @@ class GroupedDataSet[T: ClassTag](
   }
 
   /**
+    * Applies a special case of a reduce transformation (maxBy) on a grouped {@link DataSet}.<br>
+    * The transformation consecutively calls a {@link ReduceFunction}
+    * until only a single element remains which is the result of the transformation.
+    * A ReduceFunction combines two elements into one new element of the same type.
+    *
+    * @param fields Keys taken into account for finding the minimum.
+    * @return A { @link ReduceOperator} representing the minimum.
+    */
+  def maxBy(fields: Array[Int]) : ReduceOperator[T]  = {
+    if (!set.javaSet.getType().isTupleType) {
+      throw new InvalidProgramException("DataSet#maxBy(int...) only works on Tuple types.")
+    }
+    new ReduceOperator[T](
```

Comment:
Do not create a Java API `ReduceOperator` here. Use `reduce(new SelectByMaxFunction(...))` instead which creates a Scala API class.